design/aws-ecs-scheduled-ecs-task-construct.md

```diff
@@ -0,0 +1,96 @@
+
+Note: Currently, ScheduledTasks, via CloudFormation, are only supported for EC2 and not for Fargate. CloudFormation does not support every [EcsParameter](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-events-rule-ecsparameters.html) that is supported by the [Amazon CloudWatch Events API](https://docs.aws.amazon.com/AmazonCloudWatchEvents/latest/APIReference/API_EcsParameters.html). It is currently missing the following custom parameters:
+
+* Group
+* LaunchType
+* NetworkConfiguration
+* PlatformVersion
+
+TaskDefinitions contain a NetworkMode, and EC2 can support none, bridge, host or awsvpc (which requires a NetworkConfiguration attribute). However, TaskDefinitions running on Fargate can only support a NetworkMode of awsvpc. For this construct, we're creating a scheduled task (that is standalone task, not backed with a service). It works for ECS because we're using bridge as the NetworkMode, which does not require a NetworkConfiguration to be present.
+
+## General approach
+
+The new [`ecs.ScheduledEc2Task`] class will include an L3 construct for:
+
+* ScheduledEc2Task
+
+A `ScheduledEc2Task` will create a task definition with the specified container. An `Ec2EventRuleTarget` will be created and associated as the target to an `Amazon Cloudwatch Event Rule` (indicating how frequently the task should be run). Based on the `Amazon Cloudwatch Event Rule` schedule, a task will run on the EC2 instances specified in the cluster. 
+
+## Code changes
+
+Given the above, we should make the following changes to support scheduled tasks on ECS:
+1. Create `ScheduledEc2TaskProps` interface  and `ScheduledEc2Task` construct
+
+# Part 1: Create `ScheduledEc2TaskProps` interface  and `ScheduledEc2Task` construct  
+
+The `ScheduledEc2TaskProps` interface will contain properties to construct the Ec2TaskDefinition, Ec2EventRuleTarget and EventRule:
+
+```ts
+export interface ScheduledEc2TaskProps {
+  /**
+   * The cluster where your service will be deployed.
+   */
+  readonly cluster: ICluster;
+
+  /**
+   * The image to start.
+   */
+  readonly image: ContainerImage;
+
+  /**
+   * Number of desired copies of running tasks.
+   *
+   * @default 1
+   */
+  readonly desiredTaskCount?: number;
+
+  /**
+   * The hard limit (in MiB) of memory to present to the container.
+   *
+   * If your container attempts to exceed the allocated memory, the container
+   * is terminated.
+   *
+   * @default 256
+   */
+  readonly memoryLimitMiB?: number;
+
+  /**
+   * The schedule or rate (frequency) that determines when CloudWatch Events
+   * runs the rule. For more information, see Schedule Expression Syntax for
+   * Rules in the Amazon CloudWatch User Guide.
+   *
+   * @see http://docs.aws.amazon.com/AmazonCloudWatch/latest/events/ScheduledEvents.html
+   */
+  readonly scheduleExpression?: string;
+}
```

Comment:
Missing overrides in this interface. Overrides are pretty extensively used according to the Twitter thread I started yesterday: https://twitter.com/nathankpeck/status/1120719301349126145

Reply:
I think overrides is not the right term and led to some confusion.  *Settings* are missing from the props.  This construct will create the task def, so any settings in these props will go right into the task def, no overrides needed.

That said, looking at the Tweet thread, environment variables and command are important to add.

Reply:
Added environment variables, command, cpu and memoryReservationMiB to the taskdef.